ui/raidboss/emulator/EventBus.ts

```diff
@@ -5,7 +5,7 @@ type Scope = object;
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 type Param = any;
 
-type CallbackFunction = (...args: Param) => void;
+type CallbackFunction = ((...args: Param) => void) | ((...args: Param) => Promise<void>);
```

Comment:
```suggestion
type CallbackFunction = (...args: Param) => void | Promise<void>;
```

Simplification suggestion.

Reply:
I wasn't sure where `|` operator priority was for type defs here, e.g. which of the following that evaluates to:
```ts
type CallbackFunction = (...args: Param) => (void | Promise<void>);
// or
type CallbackFunction = ((...args: Param) => void) | (Promise<void>);
```

If it's the former, maybe for clarity it makes sense to wrap the return types in parenthesis as in my example?

Reply:
Changed to this:
```ts
type CallbackFunction = (...args: Param) => (void | Promise<void>);
```

Reply:
I am pretty sure that `=>` binds more strongly than `|`.  If you do something like `() => void | () => boolean` tsc will complain at you with `Function type notation must be parenthesized when used in a union type.(1385)` because it looks like it's `() => void | boolean` but is really `() => (void | (() => boolean))`.